Community/Hysteresis/automation/lib/python/community/hysteresis.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+"""
+Author: Rich Koshak
+The Hysteresis module provides an implementation that will tell you whether a
+value is below, above, or inbetween a target value plus or minus an offset.
+
+Functions
+========
+    - hysteresis: Returns a value indicating if the value is above, below, or 
+      within the hysteresis gap.
+
+License
+========
+Copyright (c) 2019 Richard Koshak
+
+This program and accomanying materials are made available under the terms of 
+the Eclipse Public License 2.0 which is available at 
+http://www.eclipse.org/legal/epl-2.0
+
+"""
+from org.eclipse.smarthome.core.library.types import QuantityType, DecimalType, PercentType
+
+@log_traceback
+def hysteresis(target, value, low=0, high=0):
+    """
+    Checks if the value is below, above or between the hysteresis gap defined
+    by the target-low <= value <= target+high. The function accepts Python 
+    primitives, QuantityTypes, DecimalTypes, or PercentTypes or any combination
+    of the four types. When using QuantityTypes, the default units of the Units
+    of Measure is used.
+
+    Arguments:
+        - target: The target value that defines the setpoint for the comparison.
+        - value: The value to determine where it is in the hysteresis.
+        - low: Value subtracted from target to define the lower bounds of the 
+        hysteresis gap. Defaults to 0.
+        - high: Value added to target to define the upper bounds of the 
+        hystersis gap. Defaults to 0
+    Returns:
+        - 1 if value is >= target+high
+        - 0 if the value is between target-low and target+high or if low and 
+        high are both zero and value == target
+        - (-1) if value is <= target-low
+    """
+    def get_float(value):
+        """Helper function to normalize all the argumets to primitive floats."""
+        if isinstance(value, (QuantityType, DecimalType, PercentType)):
+            value = value.floatValue()
+        return value
+
+    target = get_float(target)
+    value  = get_float(value)
+    low    = get_float(low)
+    high   = get_float(high)
```

Comment:
Get rid of the function and use variable unpacking with list comprehension...
```
target, value, low, high = [x.floatValue() if isinstance(x, (QuantityType, DecimalType, PercentType)) else x for x in [target, value, low, high]]
```

Reply:
I believe all the comments have been addressed with the latest set of commits.

Reply:
Do I mark the conversation resolved when I think it's been addressed or do you mark it when you confirm I've addressed it?

Reply:
I've seen it done both ways :smile:. Personally, I prefer to resolve them myself, but I always end up opening the resolved ones up to verify them before merging any how.